molgenis-data/src/main/java/org/molgenis/data/meta/model/EntityType.java

```diff
@@ -143,6 +127,44 @@ public static EntityType newInstance(
     return entityTypeCopy;
   }
 
+  /**
+   * Deep copy all attributes of an EntityType to its copy. Returns a LinkedMap of the old attribute
+   * IDs to the newly copied Attributes.
+   *
+   * @param entityType original entity meta data
+   * @param entityTypeCopy copy of entity meta data
+   * @param attrFactory attribute factory used to create new attributes in deep-copy mode
+   * @return map of old attributes IDs to copied attributes
+   */
+  public static Map<String, Attribute> deepCopyAttributes(
+      EntityType entityType, EntityType entityTypeCopy, AttributeFactory attrFactory) {
+    Map<String, Attribute> copiedAttributes = newLinkedHashMap();
```

Comment:
new LinkedHashMap<>();

From the docs:
   * <p><b>Note for Java 7 and later:</b> this method is now unnecessary and should be treated as
   * deprecated. Instead, use the {@code ArrayList} {@linkplain ArrayList#ArrayList() constructor}
   * directly, taking advantage of the new <a href="http://goo.gl/iz2Wi">"diamond" syntax</a>.